spec.html

```diff
@@ -2,20 +2,490 @@
+            _targetFunction_: a function object,
+            _snapshot_: a List of Records with fields [[AsyncContextKey]] (a Symbol) and [[AsyncContextvalue]] (an ECMAScript language value),
+          ): either a normal completion containing a function object or a throw completion
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It is used to specify the creation of new AsyncContext wrapped function exotic objects.</dd>
+        </dl>
+        <emu-alg>
+          1. Let _proto_ be ? <emu-meta effects="user-code">_targetFunction_.[[GetPrototypeOf]]</emu-meta>().
+          1. Let _internalSlotsList_ be the list-concatenation of « [[Prototype]], [[Extensible]] » and the internal slots listed in <emu-xref href="#table-internal-slots-of-asynccontext-wrapped-function-exotic-objects"></emu-xref>.
+          1. Let _obj_ be MakeBasicObject(_internalSlotsList_).
+          1. Set _obj_.[[Prototype]] to _proto_.
+          1. Set _obj_.[[Call]] as described in <emu-xref href="#sec-asynccontext-wrapped-function-exotic-objects-call-thisargument-argumentslist"></emu-xref>.
+          1. If IsConstructor(_targetFunction_) is *true*, then
+            1. Set _obj_.[[Construct]] as described in <emu-xref href="#sec-asynccontext-wrapped-function-exotic-objects-construct-argumentslist-newtarget"></emu-xref>.
+          1. Set _obj_.[[WrappedTargetFunction]] to _targetFunction_.
+          1. Set _obj_.[[AsyncContextSnapshot]] to _snapshot_.
+          1. Return _obj_.
+        </emu-alg>
+      </emu-clause>
+    </emu-clause>
+  </emu-clause>
+</emu-clause>
+
+<emu-clause id="sec-control-abstraction-objects">
+  <h1>Control Abstraction Objects</h1>
+  <emu-clause id="sec-asynccontext-objects">
+    <h1>AsyncContext Objects</h1>
+
+    <emu-clause id="sec-asynccontext-abstract-operations">
+      <h1>AsyncContext Abstract Operations</h1>
+      <emu-clause id="sec-asynccontext-snapshot" type="abstract operation">
+        <h1>
+          AsyncContextSnapshot (
+          ): a List of Records with fields [[AsyncContextKey]] (a Symbol) and [[AsyncContextvalue]] (an ECMAScript language value)
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It is used to snapshot the surrounding agent's Agent Record's [[AsyncContextMapping]].</dd>
+        </dl>
+        <emu-alg>
+          1. Let _agentRecord_ be the surrounding agent's Agent Record.
+          1. Let _asynccontextMapping_ be _agentRecord_.[[AsyncContextMapping]].
+          1. Let _snapshot_ be a new empty List.
+          1. For each Record { [[AsyncContextKey]], [[AsyncContextValue]] } _p_ of _asynccontextMapping_, do
+            1. If _p_.[[AsyncContextKey]] is not ~empty~, then
+              1. Let _q_ be the Record { [[AsyncContextKey]]: _p_.[[AsyncContextKey]], [[AsyncContextValue]]: _p_.[[AsyncContextValue]] }.
+              1. Append _q_ to _snapshot_.
+          1. Return _snapshot_.
```

Comment:
I don't think this should be closing the mapping, and the only 2 callers will be faster if they can store the reference to the mapping directly.